### DISCUSSION TITLE
How to change the magnetic field in undulator？

### DISCUSSION THREAD

**Discussion by astroquantum** [svenreiche/Genesis-1.3-Version4#115] [Q&A] [answered]
I would like to ask, if I want to change the transverse electromagnetic field distribution in the undulator, for example to a centrosymmetric electric field, how do I do it, do I need to change the source file or just change it in the input file? Thanks!

**Comment by svenreiche** [comment]
Cn you explain it a little more because I do not understand what you want. Are you referring to replacing the magnetic field of th eundulator iaht a RF field driven “wiggler”. Or are you referring to the radiation field?


Best regards,
Sven

On 18 Oct 2023, at 21:53, astroquantum ***@***.***> wrote:



I would like to ask, if I want to change the transverse electromagnetic field distribution in the undulator, for example to a centrosymmetric electric field, how do I do it, do I need to change the source file or just change it in the input file? Thanks!

**Reply by astroquantum** [reply]
Thanks for your reply. I may have made some mistakes in my wording. I am referring to replacing the magnetic field of the undulator and let the electrons move in my given electric or magnetic field. Thanks again.

**Comment by astroquantum** [comment]
Thanks for your reply. I may have made some mistakes in my wording. I am referring to replacing the magnetic field of the undulator and let the electrons move in my given electric or magnetic field. Thanks again.
发件人: Sven Reiche
发送时间: 2023年10月21日 1:00
收件人: svenreiche/Genesis-1.3-Version4
抄送: astroquantum; Author
主题: Re: [svenreiche/Genesis-1.3-Version4] How to change the transverse electromagnetic field ？ (Discussion #115)


Cn you explain it a little more because I do not understand what you want. Are you referring to replacing the magnetic field of th eundulator iaht a RF field driven “wiggler”. Or are you referring to the radiation field? 


Best regards, 
Sven 

On 18 Oct 2023, at 21:53, astroquantum ***@***.***> wrote: 



I would like to ask, if I want to change the transverse electromagnetic field distribution in the undulator, for example to a centrosymmetric electric field, how do I do it, do I need to change the source file or just change it in the input file? Thanks!

**Comment by svenreiche** [comment, marked as answer]
I am now back from my vacation and can answer more.

Genesis uses precalculated trajectories for the source term of the field equation.
Name it assumes that the transverse velocity is given as v_x = c * K/gamma sin(k_u z)

So if you want to do a laser field instead than one has to calculate the effective value of K which is as well the normalized vector potential of the radiation field.

However there is one problem, which is the transverse dependence of the undulator field.

For a magnetic undulator one has to satisfy MAxwell equation which ends up with a strong transverse dependence of the undulator field.  This is described by the roll off parameters kx and ky.

For an electric magnetic field this is not the case. It is the best to set those to kx = ky = 0 for disabling the natural focusing of the undulator field


Hope this helps





-----------------------------------

Paul Scherrer Institut

Sven Reiche
WBGB/119
CH-5232 Villigen PSI

Telefon: +41 56 310 51 19
E-Mail: ***@***.******@***.***>

________________________________
